src/ck-ldb/lbdb.h

```diff
@@ -139,12 +141,16 @@ class LBObjUserData {
     if (d.data != NULL) {
       if (data==NULL) init();
       memcpy(data, d.data, CkpvAccess(lbobjdatalayout).size());
+    } else if (data != NULL) {
+      delete [] data;
+      data = NULL;
     }
     return *this;
   }
-  inline void init() { data = new char[CkpvAccess(lbobjdatalayout).size()]; }
   inline void pup(PUP::er &p);
   void *getData(int idx) { if (data==NULL) init(); return (void*)(data+idx); }
+private:
+  inline void init() { data = new char[CkpvAccess(lbobjdatalayout).size()]; }
```

Comment:
I think this needs to be public, otherwise there's seemingly no way for a user to ever allocate space for data because the only non-copy constructor only sets `data` to NULL.

Reply:
init() is called from getData()

Reply:
Ah, I missed that, looks fine.